thoth/prescriptions_refresh/handlers/pypi_maintainers.py

```diff
@@ -0,0 +1,275 @@
+        f"{_PYPI_MAINTAINER_PROJECTS_COUNT} projects on PyPI",
+        nonexisting_ok=True,
+    )
+
+
+def _get_maintainer_join_date(maintainer: str) -> Optional[datetime.datetime]:
+    """Parse information when the given maintainer joined PyPI."""
+    response = requests.get(f"https://pypi.org/user/{maintainer}/")
+    if response.status_code != 200:
+        _LOGGER.error(
+            "Failed to obtain user page from PyPI for %r (%d): %s", maintainer, response.status_code, response.text
+        )
+        return None
+
+    soup = BeautifulSoup(response.text, "html.parser")
+    for metadiv in soup.find_all("div", class_="author-profile__metadiv"):
+        if metadiv.time and metadiv.time.get("datetime"):
+            dt: datetime.datetime = datetime_parser.parse(metadiv.time.get("datetime"), ignoretz=True)
+            return dt
+
+    return None
+
+
+def _do_maintainer_joined_warning(prescriptions: "Prescriptions", project_name: str, maintainers: List[str]) -> None:
+    """Warn if a project is maintained by maintainers that joined recently."""
+    today = datetime.datetime.today()
+    for maintainer in maintainers:
+        maintainer_join_date = _get_maintainer_join_date(maintainer)
+        if maintainer_join_date is None:
+            # This usually happens for maintainers that are on PyPI for some time. Recent users have datetime assigned.
+            # Mind that if the datetime is not parsed for recent users, join date parsing logic needs to be adjusted.
+            _LOGGER.warning("No join date parsed for maintainer %r, assuming old maintainer on PyPI", maintainer)
+            break
+
+        if (today - maintainer_join_date).days >= _PYPI_MAINTAINER_JOINED_AGE_DAYS:
+            _LOGGER.info(
+                "Project %r is maintainer by %r who was registered on PyPI more than %d days ago",
+                project_name,
+                maintainer,
+                _PYPI_MAINTAINER_JOINED_AGE_DAYS,
+            )
+            break
+    else:
+        # All maintainers joined PyPI recently.
+        prescriptions.create_prescription(
+            project_name=project_name,
+            prescription_name=_PYPI_PROJECT_MAINTAINER_JOINED_AGE_PRESCRIPTION_NAME,
+            content=_PYPI_PROJECT_MAINTAINER_JOINED_AGE_PRESCRIPTION_CONTENT.format(
+                package_name=project_name,
+                prescription_name=prescriptions.get_prescription_name("PyPIMaintainersJoinWrap", project_name),
+                maintainers_join_days=_PYPI_MAINTAINER_JOINED_AGE_DAYS,
+            ),
+            commit_message=f"Project {project_name!r} has maintainers who joined PyPI recently",
+        )
+        return
+
+    # A trusted maintainer found.
+    prescriptions.delete_prescription(
+        project_name,
+        _PYPI_PROJECT_MAINTAINER_JOINED_AGE_PRESCRIPTION_NAME,
+        commit_message=f"Project {project_name!r} has a maintainers who are on PyPI for a longer period of time",
```

Comment:
```suggestion
        commit_message=f"Project {project_name!r} has a maintainer who is on PyPI for a longer period of time",
```

Reply:
Actually, the plural is right here. I've adjusted the message. Thanks!